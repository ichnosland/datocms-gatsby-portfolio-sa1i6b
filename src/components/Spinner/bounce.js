import styled, { css, keyframes } from 'styled-components';

import { colore } from 'style/color';
import media from 'style/mediainjector';

export const delay = (turn = 0.0) => ` -${turn * 0.16}s`;

const bounce = keyframes`
  0%, 80%, 100% {
    -webkit-transform: scale(0);
    transform: scale(0);
  } 40% {
    -webkit-transform: scale(1.0);
    transform: scale(1.0);
  }
`;

export const dimension = (size) => {
  switch (size) {
    case 'small':
      return '10px';
    case 'huge':
      return '30px';
    default:
      return '18px';
  }
};

export const Bounce = styled.div`
  width: ${(props) => dimension(props.size)};
  height: ${(props) => dimension(props.size)};
  background-color: ${(props) => props.bgColor ? props.bgColor : props.theme.brand};
  ${({ inverse }) => inverse && css`background-color: ${colore.ui.contrast};`}
  ${media.mobile`
    ${(props) => props.ringo && css`
      background-color: ${colore.ui.contrast};
    `};
  `};
  border-radius: 100%;
  display: inline-block;
  animation: ${bounce} 1.4s infinite ease-in-out both;
  animation-delay: ${(props) => props.number ? delay(props.number) : 0};
`;

Bounce.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};
