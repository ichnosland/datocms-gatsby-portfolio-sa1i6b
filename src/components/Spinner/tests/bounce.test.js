import React from 'react';
import { shallow } from 'enzyme';

import { Bounce, dimension, delay } from '../bounce';

describe('delay ', () => {
  it('should work w/o args', () => {
    const del = delay();
    expect(del).toEqual(' -0s');
  });

  it('should work w/ 0', () => {
    const del = delay(0);
    expect(del).toEqual(' -0s');
  });

  it('should work w/ 1', () => {
    const del = delay(1);
    expect(del).toEqual(' -0.16s');
  });

  it('should work w/ 2', () => {
    const del = delay(2);
    expect(del).toEqual(' -0.32s');
  });
});

describe('dimension ', () => {
  it('should work w/o args', () => {
    const size = dimension();
    expect(size).toEqual('18px');
  });

  it('should work w/ unused args', () => {
    const size = dimension('unused');
    expect(size).toEqual('18px');
  });

  it('should work w/ small args', () => {
    const size = dimension('small');
    expect(size).toEqual('10px');
  });

  it('should work w/ huge args', () => {
    const size = dimension('huge');
    expect(size).toEqual('30px');
  });
});

describe('<Bounce />', () => {
  it('should render a <div> tag', () => {
    const renderedComponent = shallow(<Bounce />);
    expect(renderedComponent.type()).toEqual('div');
  });

  it('should render its css', () => {
    const renderedComponent = shallow(<Bounce />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props theme', () => {
    const theme = {
      brand: 'red',
    };
    const renderedComponent = shallow(<Bounce theme={theme} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props bgColor', () => {
    const color = 'blue';
    const renderedComponent = shallow(<Bounce bgColor={color} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props inverse', () => {
    const renderedComponent = shallow(<Bounce inverse />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props ringo', () => {
    const renderedComponent = shallow(<Bounce ringo />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props size small', () => {
    const size = 'small';
    const renderedComponent = shallow(<Bounce size={size} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props size huge', () => {
    const size = 'huge';
    const renderedComponent = shallow(<Bounce size={size} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props number 1', () => {
    const number = 1;
    const renderedComponent = shallow(<Bounce number={number} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props number 2', () => {
    const number = 2;
    const renderedComponent = shallow(<Bounce number={number} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
