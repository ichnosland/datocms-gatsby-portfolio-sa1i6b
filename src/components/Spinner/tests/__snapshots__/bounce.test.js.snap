// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bounce /> should render its css 1`] = `
.c0 {
  width: 18px;
  height: 18px;
  background-color: #00BBEF;
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: ipJYoS 1.4s infinite ease-in-out both;
  animation: ipJYoS 1.4s infinite ease-in-out both;
  -webkit-animation-delay: 0;
  animation-delay: 0;
}

<div
  className="c0"
/>
`;

exports[`<Bounce /> should render its css w/ props bgColor 1`] = `
.c0 {
  width: 18px;
  height: 18px;
  background-color: blue;
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: ipJYoS 1.4s infinite ease-in-out both;
  animation: ipJYoS 1.4s infinite ease-in-out both;
  -webkit-animation-delay: 0;
  animation-delay: 0;
}

<div
  className="c0"
/>
`;

exports[`<Bounce /> should render its css w/ props inverse 1`] = `
.c0 {
  width: 18px;
  height: 18px;
  background-color: #00BBEF;
  background-color: rgb(255,255,255);
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: ipJYoS 1.4s infinite ease-in-out both;
  animation: ipJYoS 1.4s infinite ease-in-out both;
  -webkit-animation-delay: 0;
  animation-delay: 0;
}

<div
  className="c0"
/>
`;

exports[`<Bounce /> should render its css w/ props number 1 1`] = `
.c0 {
  width: 18px;
  height: 18px;
  background-color: #00BBEF;
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: ipJYoS 1.4s infinite ease-in-out both;
  animation: ipJYoS 1.4s infinite ease-in-out both;
  -webkit-animation-delay: -0.16s;
  animation-delay: -0.16s;
}

<div
  className="c0"
/>
`;

exports[`<Bounce /> should render its css w/ props number 2 1`] = `
.c0 {
  width: 18px;
  height: 18px;
  background-color: #00BBEF;
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: ipJYoS 1.4s infinite ease-in-out both;
  animation: ipJYoS 1.4s infinite ease-in-out both;
  -webkit-animation-delay: -0.32s;
  animation-delay: -0.32s;
}

<div
  className="c0"
/>
`;

exports[`<Bounce /> should render its css w/ props ringo 1`] = `
.c0 {
  width: 18px;
  height: 18px;
  background-color: #00BBEF;
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: ipJYoS 1.4s infinite ease-in-out both;
  animation: ipJYoS 1.4s infinite ease-in-out both;
  -webkit-animation-delay: 0;
  animation-delay: 0;
}

@media (max-width:64em) {
  .c0 {
    background-color: rgb(255,255,255);
  }
}

<div
  className="c0"
/>
`;

exports[`<Bounce /> should render its css w/ props size huge 1`] = `
.c0 {
  width: 30px;
  height: 30px;
  background-color: #00BBEF;
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: ipJYoS 1.4s infinite ease-in-out both;
  animation: ipJYoS 1.4s infinite ease-in-out both;
  -webkit-animation-delay: 0;
  animation-delay: 0;
}

<div
  className="c0"
  size="huge"
/>
`;

exports[`<Bounce /> should render its css w/ props size small 1`] = `
.c0 {
  width: 10px;
  height: 10px;
  background-color: #00BBEF;
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: ipJYoS 1.4s infinite ease-in-out both;
  animation: ipJYoS 1.4s infinite ease-in-out both;
  -webkit-animation-delay: 0;
  animation-delay: 0;
}

<div
  className="c0"
  size="small"
/>
`;

exports[`<Bounce /> should render its css w/ props theme 1`] = `
.c0 {
  width: 18px;
  height: 18px;
  background-color: red;
  border-radius: 100%;
  display: inline-block;
  -webkit-animation: ipJYoS 1.4s infinite ease-in-out both;
  animation: ipJYoS 1.4s infinite ease-in-out both;
  -webkit-animation-delay: 0;
  animation-delay: 0;
}

<div
  className="c0"
/>
`;
